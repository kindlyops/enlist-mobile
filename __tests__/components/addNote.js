import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { fromJS } from 'immutable'
import moment from 'moment'
import AddNote from '../../app/components/application/addNote'

describe('Initial state', () => {
  test('matches snapshot', () => {
    let users = fromJS([
      {
        id: 1,
        fullName: 'Test User'
      }
    ])

    let addNote = () => {}
    let onStop = () => {}

    const tree = renderer.create(
      <AddNote addNote={addNote} users={users} onStop={onStop} />
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('adding a note and hitting submit works', () => {
    let users = fromJS([
      {
        id: 1,
        fullName: 'Test User'
      }
    ])

    let addNote = text => {
      expect(text).toEqual('Here is a note')
    }

    let onStop = () => {}

    const component = shallow(
      <AddNote addNote={addNote} users={users} onStop={onStop} />
    )
    expect(component).not.toBeNull()
    expect(component.find('KeyboardView')).toExist()

    component.find('TextInput').props().onChange({
      nativeEvent: {
        text: 'Here is a note',
        contentSize: {}
      }
    })

    component.find('Button').props().onPress()
  })

  test('adding a note with @ brings up the mentions sheet', () => {
    let users = fromJS([
      {
        id: 1,
        fullName: 'Test User',
        email: 'test@example.org'
      }
    ])

    let addNote = text => {
      expect(text).toEqual('Hi @test')
    }

    let onStop = () => {}

    const component = shallow(
      <AddNote addNote={addNote} users={users} onStop={onStop} />
    )
    expect(
      component.findWhere(node => node.prop('testID') === 'add-note-users')
    ).not.toExist()

    component.find('TextInput').props().onChange({
      nativeEvent: {
        text: 'Hi @test',
        contentSize: {}
      }
    })

    expect(
      component.findWhere(node => node.prop('testID') === 'add-note-users')
    ).toExist()
  })
})
