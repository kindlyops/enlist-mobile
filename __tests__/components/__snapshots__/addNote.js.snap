// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial state matches snapshot 1`] = `
<Modal
  animationType="fade"
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <ScrollView
    contentContainerStyle={
      Array [
        Object {
          "justifyContent": "flex-end",
        },
        Object {
          "height": 1334,
        },
      ]
    }
    keyboardShouldPersistTaps={true}
    scrollEnabled={false}
    style={
      Object {
        "backgroundColor": "rgba(245,248,251,.99)",
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FCFCFC",
          "borderTopColor": "#EEEEEE",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <TextInput
        autoFocus={true}
        multiline={true}
        onChange={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Add a note..."
        style={
          Array [
            Object {
              "backgroundColor": "#FCFCFC",
              "borderColor": "#EEEEEE",
              "borderRadius": 4,
              "borderWidth": 1,
              "fontSize": 13,
              "marginLeft": 10,
              "marginRight": 10,
              "padding": 8,
            },
            Object {
              "flex": 1,
              "height": 40,
            },
          ]
        }
        value=""
      />
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4285F4",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontWeight": "500",
              },
              undefined,
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </ScrollView>
</Modal>
`;
